lib/stimulus_reflex/channel.rb

```diff
@@ -35,7 +35,9 @@ def receive(data)
     arguments = (data["args"] || []).map { |arg| object_with_indifferent_access arg }
     element = StimulusReflex::Element.new(data)
     permanent_attribute_name = data["permanent_attribute_name"]
+    form_data = Rack::Utils.parse_nested_query(data["form_data"] || "")
     params = data["params"] || {}
+    params.deep_merge!(form_data)
```

Comment:
I would suggest changing the order of merging to allow params overwriting:
```suggestion
    params = Rack::Utils.parse_nested_query(data["form_data"] || "")
    params.deep_merge!(data["params"] || {})
```

Here is focused example where param `foo` is explicitly set in stimulus controller:

view:
```html
<form>
  <input name="foo" type="text" value="chicken">
  <div id="result"></div>
  <a data-action="chicken-egg#poke" data-controller="chicken-egg" href="#">Poke!</a>
</form>
```
stimulus controller:
```javascript
import { Controller } from 'stimulus'
import StimulusReflex from 'stimulus_reflex'

export default class extends Controller {
  connect() {
    StimulusReflex.register(this)
  }

  beforeReflex(element) {
    const { params } = element.reflexData
    element.reflexData.params = { ...params, foo: 'egg' }
  }

  poke(event) {
    event.preventDefault()
    this.stimulate('ChickenEgg#poke')
  }
}
```
reflex:
```ruby
class ChickenEggReflex < StimulusReflex::Reflex
  def poke
    morph '#result', params[:foo]
  end
end
```

Reply:
This makes sense - as you already described this in #325. 
I changed it accordingly.

Reply:
Great! Thank you!